src/pages-and-resources/discussions/DiscussionsSettings.jsx

```diff
@@ -31,6 +32,13 @@ function DiscussionsSettings({ courseId, intl }) {
   const dispatch = useDispatch();
   const { path: pagesAndResourcesPath } = useContext(PagesAndResourcesContext);
   const { status, hasValidationError } = useSelector(state => state.discussions);
+  const courseDetail = useModel('courseDetails', courseId);
+
+  let canChangeProviders = false;
+  if (courseDetail) {
+    const courseStarted = new Date(courseDetail.start) <= new Date();
+    canChangeProviders = getAuthenticatedUser().administrator || !courseStarted;
+  }
```

Comment:
This code seems to be executing every time the component is rendered. 

I am very confused if this is the right place for these lines. My concern is, this is not something which will change for the current user and course and calculating it over every render seems redundant. I'd move this code to somewhere it only updates when the API call is made to update course details `courseDetails`. Can you think of any such way?

Reply:
I can put this inside a [useMemo hook](https://reactjs.org/docs/hooks-reference.html#usememo)...
I don't think it'll be much of an improvement though: this isn't a computation I'd consider to be expensive or, at least, more expensive than whatever machinery `userMemo` employs.

Let me know if you still want me to change it though.